### PR TITLE
dev/sg: update 'sg rfc' with new RFC terms, modernize command

### PR DESCRIPTION
**Pull request by bobheadxi**
Was playing around with `sg rfc` with @burmudar and noticed that many of the new RFC modifiers (`PRIVATE`, `DONE`) are not supported, and we also have this nil pointer being rendered for unknown modifiers (often misspellings).

This extends `sg rfc` to support the new terms and the `PRIVATE` modifier, and also modernizes the `sg rfc` command so that the subcommands are real subcommands and thus render in docs, support help flags, and support autocompletions.

## Test plan

`go run ./dev/sg rfc -h`

`go run ./dev/sg rfc search -h`

`go run ./dev/sg rfc list`

image

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #41579** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/41579" target="_blank">Graphite</a> 
    * **PR #42893** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/42893" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/42893?utm_source=stack-comment).

**Review by burmudar — APPROVED** [review]
Ahhh man. This is so nice! Thanks

**Review by burmudar — CHANGES_REQUESTED** [review]
@bobheadxi did you mean to merge into the gcloud secrets PR? Main has some other changes which splits the rfc into beong able to search public and private rfcs

**Review by michaellzc — APPROVED** [review]
lgtm

**Comment by michaellzc** [comment]
unrelated comment regarding search result quality

```sh
go run ./dev/sg rfc search 'private terraform' | wc -l                                                                                                                                                                                                                  10:53:12
     108
```

looks like `queryRFC` is trying to match stuff from both title and body , so there is a lot of noise

**Comment by bobheadxi** [comment]
> did you mean to merge into the gcloud secrets PR? Main has some other changes which splits the rfc into beong able to search public and private rfcs

Yes, I didn't want to conflict with your changes - I've stacked this PR on top of yours :)

**Comment by bobheadxi** [comment]
> looks like queryRFC is trying to match stuff from both title and body , so there is a lot of noise

Yeah a next step could be to configure the query with flags (`--title` maybe, and I think a `description` field is also available - I see in one of your docs @michaellzc you have `#cloud` in it, but that's the only usage I've seen)

**Comment by michaellzc** [comment]
> > looks like queryRFC is trying to match stuff from both title and body , so there is a lot of noise
> 
> Yeah a next step could be to configure the query with flags (`--title` maybe, and I think a `description` field is also available - I see in one of your docs @michaellzc you have `#cloud` in it, but that's the only usage I've seen)

I was playing around with the search in GDrive and see if it can pick up stuff in the doc description

 Label is actually a separate feature and needs #it-tech-ops to enable it https://support.google.com/a/answer/9292382?hl=en

**Review by mrnugget — APPROVED** [review]

**Review by burmudar — APPROVED** [review]

File: dev/sg/internal/rfc/rfc.go [hunk 1]
```diff
@@ -22,26 +22,26 @@ import (
...
-	ParentID:    "1zP3FxdDlcSQGC1qvM9lHZRaHH4I9Jwwa",
+	FolderID:    "1zP3FxdDlcSQGC1qvM9lHZRaHH4I9Jwwa",
 	OrderBy:     "createdTime,name",
 }
 
 var PrivateDrive = DriveSpec{
 	DisplayName: "Private",
 	DriveID:     "0AK4DcztHds_pUk9PVA", // Sourcegraph DriveID
-	ParentID:    "1KCq4tMLnVlC0a1rwGuU5OSCw6mdDxLuv",
+	FolderID:    "1KCq4tMLnVlC0a1rwGuU5OSCw6mdDxLuv",
```
Comment (bobheadxi) [inline dev/sg/internal/rfc/rfc.go:32]: cc @sanderginn I think this is folder ID - using this for the link (see the new generated `Description`) points to the folder, I'm guessing the confusion comes from using it in the `parents in '%s'` query, which says "look for files with parent $ID", which in this case is a folder ID

**Pull request merged**